### PR TITLE
[AD-258] Create HdAddress internally within createAddress

### PR DESCRIPTION
**Pull request by epicallan**
## Description

- Create HdAddress internally within createAddress in Ariadne.Wallet.Cardano.Kernel.DB.AcidState
- Split createAccPrefiltered into 2 functions one of which is createAddrPrefiltered
- Move Creation logic from Ariadne.Wallet.Cardano.Kernel.DB.AcidState to  Ariadne.Wallet.Cardano.Kernel.DB.HdWallet.Create

## Related issue(s)

https://issues.serokell.io/issue/AD-258

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - [TUI usage guide](../tree/master/docs/usage-tui.md)
    - [GUI usage guide](../tree/master/docs/usage-gui.md)
    - [Configuration documentation](../tree/master/docs/configuration.md)
    - [GUI architecture documentation](../tree/master/docs/gui-architecture.md)
    - Haddock

- Public contracts
  - [x] Any modifications of public contracts comply with the [Evolution
  of Public Contracts](https://www.notion.so/serokell/Evolution-of-Public-Contracts-2a3bf7971abe4806a24f63c84e7076c5) policy.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Comment by gromakovsky** [comment]
From PR template:
```
Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

If you don't set a checkmark (e. g. don't add a test for new functionality),
you must be able to justify that.
```

**Comment by epicallan** [comment]
@gromakovsky I have updated checkmarks.

File: ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/DB/AcidState.hs [hunk 1]
```diff
@@ -426,23 +446,18 @@ createAccPrefiltered mkPrefilteredUtxo accApplyP narrowP addrApplyP accId p mbAc
...
-createHdAddress hdAddress = runUpdate' . zoom dbHdWallets $
+createHdAddress :: HdAddressId
+                -> Core.Address
+                -> AddrCheckpoint
+                -> Update DB (Either HD.CreateHdAddressError HdAddress)
+createHdAddress addressId newAddress firstCheckpoint = runUpdate' . zoom dbHdWallets $ do
+    let hdAddress :: HdAddress
+        hdAddress = HD.initHdAddress addressId (InDb newAddress) firstCheckpoint
     HD.createHdAddress hdAddress
+    pure hdAddress
```
Comment (gromakovsky) [inline ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/DB/AcidState.hs]: The idea was to use `createAddPrefiltered` in this function. Currently this refactoring doesn't make much sense, you added a subfunction of `createAccPrefiltered` and use it only in `createAccPrefiltered`.

File: ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Addresses.hs [hunk 2]
```diff
@@ -168,15 +167,14 @@ createHdSeqAddress passphrase esk accId chain pw = runExceptT $ go 0
             (CreateAddressHdSeqGenerationFailed accId)
             mbAddr
 
-        let hdAddress = initHdAddress hdAddressId (InDb newAddress) firstCheckpoint
-        res <- liftIO $ update db (CreateHdAddress hdAddress)
+        res <- liftIO $ update db (CreateHdAddress hdAddressId newAddress firstCheckpoint)
```
Comment (gromakovsky) [inline ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Addresses.hs]: You should't pass `firstCheckpoint` there. `createHdAddress` can do it internally.

File: ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/DB/HdWallet/Create.hs [hunk 3]
```diff
@@ -6,28 +6,36 @@ module Ariadne.Wallet.Cardano.Kernel.DB.HdWallet.Create
...
        , CreateHdAccountError(..)
        , CreateHdAddressError(..)
-        -- * Functions
+         -- * Functions
+       , createAddrPrefiltered
        , createHdRoot
        , createHdAccount
        , createHdAddress
+       , createPrefiltered
+         -- * utilities
```
Comment (Pastafarianist) [inline ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/DB/HdWallet/Create.hs]: Minor: please capitalize the first letter to maintain uniformity.

**Pull request merged**